SampleApplications/Samples/XamarinClient/XamarinClient/SampleClient.cs

```diff
@@ -0,0 +1,469 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Opc.Ua;
+using Opc.Ua.Client;
+using System.Security.Cryptography.X509Certificates;
+using Xamarin.Forms;
+using System.Net;
+
+namespace XamarinClient
+{
+    public class SampleClient
+    {
+        public enum ConnectionStatus
+        {
+            None,
+            NotConnected,
+            Connected,
+            Error
+        }
+
+        public ConnectionStatus connectionStatus;
+        public bool haveAppCertificate;
+        public Session session;
+
+        private LabelViewModel info;
+        private ApplicationConfiguration config;
+
+        public SampleClient (LabelViewModel text)
+        {
+            connectionStatus = ConnectionStatus.None;
+            session = null;
+            info = text;
+            haveAppCertificate = false;
+            config = null;
+        }
+
+        public async void CreateCertificate()
+        {
+            string currentFolder = DependencyService.Get<IPathService>().PublicExternalFolder.ToString();
+
+            config = new ApplicationConfiguration()
+            {
+                ApplicationName = "UA Core Sample Client",
+                ApplicationType = ApplicationType.Client,
+                ApplicationUri = "urn:" + Dns.GetHostName() + ":OPCFoundation:CoreSampleClient",
+
+                SecurityConfiguration = new SecurityConfiguration
+                {
+                    ApplicationCertificate = new CertificateIdentifier
+                    {
+                        StoreType = "Directory",
+                        StorePath = currentFolder + "UA_MachineDefault",
+                        SubjectName = "UA Core Sample Client"
+                    },
+                    TrustedPeerCertificates = new CertificateTrustList
+                    {
+                        StoreType = "Directory",
+                        StorePath = currentFolder + "UA Applications",
+                    },
+                    TrustedIssuerCertificates = new CertificateTrustList
+                    {
+                        StoreType = "Directory",
+                        StorePath = currentFolder + "UA Certificate Authorities",
+                    },
+                    RejectedCertificateStore = new CertificateTrustList
+                    {
+                        StoreType = "Directory",
+                        StorePath = currentFolder + "RejectedCertificates",
+                    },
+                    NonceLength = 32,
+                    AutoAcceptUntrustedCertificates = true
+                },
+                TransportConfigurations = new TransportConfigurationCollection(),
+                TransportQuotas = new TransportQuotas { OperationTimeout = 15000 },
+                ClientConfiguration = new ClientConfiguration { DefaultSessionTimeout = 60000 }
+            };
```

Comment:
this is a really bad example and must be removed also from the Consoleclient. Use config files with opc.ua.configuration library instead